### PR TITLE
Add dusk style variation

### PR DESCRIPTION
**Pull request by carolinan**
<!-- Thanks for contributing to Twenty Twenty-Five! Please follow the Contributing Guidelines:
https://github.com/WordPress/twentytwentyfive/blob/trunk/README.md#contributing -->

**Description**
Add the dusk (Vollkorn + Fira code) global style variation.
Partial for https://github.com/WordPress/twentytwentyfive/issues/193

[Figma](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=77-338&m=dev)

**Known limitations**
The outline button style is missing.

**Testing Instructions**
On your WordPress test installation, add some varied sample content:
Posts with and without tags and categories
Posts with and without featured images.

Add:
- A code block, quote, pullquote, list block, and headings H1 to H6.
- Buttons, with default and outline style variation.
- Search

<details><summary>Here is sample content for testing the group blocks with the section styles:</summary>

```
<!-- wp:group {"className":"is-style-section-1","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group is-style-section-1" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:paragraph -->
<p>Section style 1 (but with extra padding...)</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="#">A standard button</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button" href="#">An outlined button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group -->

<!-- wp:group {"className":"is-style-section-2","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group is-style-section-2" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:paragraph -->
<p>Section style 2 (but with extra padding...)</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="#">A standard button</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button" href="#">An outlined button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group -->

<!-- wp:group {"className":"is-style-section-3","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group is-style-section-3" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:paragraph -->
<p>Section style 3 (but with extra padding...)</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="#">A standard button</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button" href="#">An outlined button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group -->

<!-- wp:group {"className":"is-style-section-4","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group is-style-section-4" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:paragraph -->
<p>Section style 4 (but with extra padding...)</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="#">A standard button</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button" href="#">An outlined button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group -->

<!-- wp:group {"className":"is-style-section-5","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group is-style-section-5" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:paragraph -->
<p>Section style 5 (but with extra padding...)</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="#">A standard button</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link wp-element-button" href="#">An outlined button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group -->
```

</details>

Go to Appearance > Editor > Styles
At the top of the styles panel, select "Dusk".
Save.

Compare the design with the design in Figma.

**Screenshot**


![66 local_2024_09_04_section-styles_](https://github.com/user-attachments/assets/2cffba9c-b03a-40ae-bedf-16ff35153051)

### CODE REVIEW

**Comment by beafialho** [comment]
This is looking great!

Some small details I noticed:

- All headings (including post titles, archive titles, search results...) should use color Accent 3 `#191919`
- The outline button seems smaller than the filled one and its border is a bit too thick for the font:

Captura de ecrã 2024-09-04, às 10 07 33

Can you apply border `1px` and make it the same dimensions as the filled button?

**Comment by carolinan** [comment]
The outline button needs custom CSS added by a different pull request.

**Comment by carolinan** [comment]
> * All headings (including post titles, archive titles, search results...) should use color Accent 3 `#191919`

What about when the site title and post title are links?

**Comment by beafialho** [comment]
> What about when the site title and post title are links?

Those two should use Accent 3 as well when they are links. The only links that should not use Accent 3 are links in a paragraph or post categories, tags, author.

File: styles/dusk-2.json [hunk 1]
```diff
@@ -0,0 +1,321 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/theme.json",
+	"version": 3,
+	"title": "Dusk",
+	"settings": {
+		"color": {
+			"palette": [
+				{
+					"color": "#E2E2E2",
```
Comment (kishanjasani) [inline styles/dusk-2.json]: As per figma 

Base color is: `#E5E5E5`,
contrast: `#191919`,
Accent-1: `#650DD4`,
Accent-2: `#FFFFFF`,
Primary: `#3B3B3B`

All these colors are slightly different, and some are assigned to different slugs.

Comment (carolinan) [reply styles/dusk-2.json]: The colors are the same as in the dusk color palette. 
The **names/slugs** in Figma have not been updated to match the names needed in the theme.

Comment (kishanjasani) [reply styles/dusk-2.json]: Got it. In Figma it's wrong: https://prnt.sc/k9H-Aq6-sqA7

Comment (carolinan) [reply styles/dusk-2.json]: I'll add the "needs design feedback" label.

**Comment by carolinan** [comment]
I am running into problems with the outline style for the buttons, when they are in groups that use the section styles.
The outlined buttons now have the correct size, which matches the default button style: but not the correct color.

Here is a screenshot of the two button styles, in group blocks with section styles.
The border color of the outline button is set to `contrast`
And on section style 4, only the text is visible:

image

The hover background color is also set to contrast, using the CSS file.

**Comment by carolinan** [comment]
Ok, I managed to set the border color to `currentColor`:
image

But when hovering over the button in style section 4, it still uses the contrast background color from the CSS so only the text is visible:

https://github.com/user-attachments/assets/0d51a0a5-216c-4b03-be1f-0ad7a84891d4

**Comment by carolinan** [comment]
I don't think the contrast border color looks so good on the default button variation either.

**Comment by beafialho** [comment]
Thank you @carolinan, the buttons and the variation is looking good to me overall!

> But when hovering over the button in style section 4, it still uses the contrast background color from the CSS so only the text is visible:

It's hard for me to provide feedback here, given that there seem to be some technical particularities/constraints that fall outside my knowledge. When hovering over the button in style section 4, if we can't find a better solution, that I assume it'll have to be like that?

**Comment by carolinan** [comment]
I wish we could know what style variation was active so that we could target it with the button CSS that the theme is already loading.
https://github.com/WordPress/gutenberg/issues/43405

**Comment by carolinan** [comment]
When there are many menu items, the site title wraps to two lines and it does not look very good to me.
I think the no-wrap needs to be changed back to wrap so that the title and navigation are on separate lines, instead of reducing the space available for the title:

image

The letter spacing in the navigation block in this variation could be reduced, but it wouldn't help if there were even more menu items than in this example.

**Comment by carolinan** [comment]
The color contrast in these patterns is not working.

### Link in bio heading, paragraph, links and full-height image
- The heading is not visible at all, the links have too low contrast:
image

### Landing page for event
About the event heading color has too low contrast:
image

### Event RSVP
The RSVP heading color has too low contrast:
image

### Landing page for podcast
The heading color has too low contrast.
The pills text color has too low contrast
image

**Comment by carolinan** [comment]
After link and heading color update:




page-patterns-dusk

**Pull request merged**